lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/MethodProcessor.java

```diff
@@ -527,8 +531,10 @@ public String key() {
   @Override
   protected MethodMetadataEntry toMetadataEntry(String line) {
     String[] elements = line.split(":");
-    if (elements.length != 2) throw new RuntimeException("invalid builtin metadata entry: " + line);
-    return new MethodMetadataEntry(elements[0], elements[1]);
+    if (elements.length < 2 || elements.length > 4) throw new RuntimeException("invalid builtin metadata entry: " + line);
+    boolean isStatic = elements.length >= 3 ? Boolean.valueOf(elements[2]) : false;
+    Owner owner = elements.length == 4 ? Owner.valueOf(elements[3]) : Owner.TYPE;
+    return new MethodMetadataEntry(elements[0], elements[1], isStatic, owner);
```

Comment:
This is a bit complicated. Aren't we always doing `<fullEnsoName>:<className>:<isStatic>:<owner>` now??

i.e. cannot we just assume that `elements.length == 4` always? That would make the code a bit simpler.

Reply:
Wanted to be backwards compatible but it is probably an overkill here.

Reply:
Is there any scenario where we would hold metadata from the old version?
From what I see, the IR should be invalidated, and once it is generated it's always with the new schema. So I can't imagine a scenario where old format could be encountered.

Reply:
Actually, this particular check should stay because of incremental compilation. If we merge this change, the old entries (with a smaller number of entries) will throw an exception, and will be simply discarded. On the Builtins.java we can however simplify that.

Reply:
I'm not convinced - while this isn't a very important piece of logic, I really don't like steering a piece of logic in our sofware by limitations of our build system. We can just clean the CI runners when merging this PR and all should work, right? It feels really wasteful to introduce logic into our system that *will stay for a very long time* just because our incremental compiler cannot notice the change properly (which will happen only *once* after the PR is merged)